### ISSUE TITLE
keine Aktualisierung der InputMask 

### ISSUE THREAD

**Issue by dgrohmann** [bstroebl/DataDrivenInputMask#32]
Ich nutze Data-Driven Input Mask V2.4.0 zusammen mit QGIS V3.12.1 und PostgreSQL/PostGIS 11.5.

Wenn ich ein Tabellenfeld zu einer bereits erstellten InputMask hinzufüge, werden alle erforderlichen Informationen in den Systemtabellen dd_table, dd_tab und dd_field gespeichert. Das Tabellenfeld wird jedoch nicht in der InputMask dargestellt. Das Feld field_skip ist OFF und field_enabled ist ON.

Wie erreiche ich eine Aktualisierung meiner InputMask?

Vielen Dank und beste Grüße
dgrohmann

**Comment by bstroebl** [comment]
Hallo,
ein Eintrag in dd_field erzeugt kein Datenbankfeld. Sie müssen das Feld zuerst **in der Datenbank** hinzufügen, danach können Sie es in dd_field konfigurieren. 
Wenn ich Sie falsch verstanden habe, beschreiben Sie bitte was Sie tun, was Sie erwarten, dass passiert und was stattdessen passiert.

**Comment by dgrohmann** [comment]
Hallo,

für eine Datenbank-Tabelle (QGIS-Layer) gibt es eine konfigurierte InputMask. Nun musste in der Datenbank-Tabelle ein weiteres Feld (Spalte) hinzugefügt werden. Als nächstes wurde dieses Feld mit dem Menüpunkt "konfiguriere Maske" zur bestehenden InputMask hinzugefügt.
Wenn man die InputMask aufruft, z.B. als Layer-Aktion oder auch über den Menüeintrag "Eingabemaske anzeigen" wird die InputMask mit ihrem ursprünglichen Layout angezeigt. Das neue Feld fehlt. Für die Kontrolle habe ich die Tabellen dd_table, dd_tab und dd_field überprüft. Dor wurde das neue Feld korrekt eingetragen.

**Comment by bstroebl** [comment]
Haben Sie den Layer aktualisiert bzw. neu aus der DB geladen bzw. QGIS neu gestartet? QGIS muß auch davon in Kenntnis gesetzt werden, dass der Layer in der DB ein neues Feld hat.
Welcher Art ist das Feld?

**Comment by bstroebl** [comment]
ausserdem muß die Maske bei einer Änderung der Konfiguration neu initialisiert werden. Am einfachsten ist es immer QGIS zu schließen und neu zu öffnen, dann sind alle Artefakte weg.

**Comment by dgrohmann** [comment]
> Haben Sie den Layer aktualisiert bzw. neu aus der DB geladen bzw. QGIS neu gestartet? QGIS muß auch davon in Kenntnis gesetzt werden, dass der Layer in der DB ein neues Feld hat.

Ja, der Layer wurde sowohl manuell (Layer entfernen, Layer hinzufügen) aktualisiert als auch mittels QGIS-Beenden und QGIS-Neustart. Beides brachte nicht den gewünschtel Erfolg.

> Welcher Art ist das Feld?

ein Datumsfeld (Datentyp: date)

**Comment by bstroebl** [comment]
schwierig.... ändern Sie doch bitte den Tabellennamen in dd_table, so dass der Layer keine Konfiguratin mehr hat, initialisieren Sie dann neu und sehen Sie bitte nach, ob das Feld in der Maske vorhanden ist.
Könnten Sie die DDL der Tabelle hier posten?

**Comment by dgrohmann** [comment]
In der Tabelle dd_table wurde der Tabellenname geändert. Anschließend QGIS beendet, neu gestartet, leeres Projekt angelegt, PostGIS-Layer "point" dem projekt hinzugefügt, Layer initialisiert, neue Maske konfiguriert ... Das Feld "erfassungszeit" erscheint nicht in der Maske.  

hier die DDL der Tabelle:
CREATE TABLE point (
	gid int4 NOT NULL,
	the_geom geometry(POINT, 25832),
	erfassungszeit timestamp DEFAULT '1900-01-01 00:00:00'::timestamp without time zone,
	CONSTRAINT point_pkey PRIMARY KEY (gid)
);
Ich melde mich morgen wieder.
Vielen Dank schon einmal.

**Comment by bstroebl** [comment]
timestamp wird aktuell von DDIM nicht unterstützt, unterstützt wird nur date, siehe [wiki](https://github.com/bstroebl/DataDrivenInputMask/wiki/Simple-Widgets)

**Comment by bstroebl** [comment]
Hallo, ist die Frage damit gelöst?

**Comment by dgrohmann** [comment]
Guten Morgen,
ich war gestern anderweitig gebunden und konnte daher nicht auf ihre Antwort reagieren.
Das mit timestamp ist mir vollkommen durchgerutscht. Nachdem ich das timestampfeld-deaktiviert und seinen Feldstatus auf "to skip" gesetzt habe, funktioniert die Inputmaske wieder. Vielen Dank für den Hinweis!

Anscheinend war dies die Ursache. An dieser Stelle hätte ich gern eine Fehlermeldung gehabt ;-)

BG

**Comment by bstroebl** [comment]
Was würden Sie erwarten? Die Maske funktioniert ja grundsätzlich, allerdings gibt es in der Tabelle ein Feld, das nicht angezeigt werden kann, weil sein Typ nicht unterstützt wird. Wenn dann beim initialisieren stets ein Fehler oder Hinweis angezeigt wird, wäre das IMHO nicht sinnvoll, denn das Feld könnte ja auch mit Absicht da sein. Deswegen ist es auch kein Fehler.
